src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/ConnectionPoolTest/ConnectionPoolTest.cs

```diff
@@ -270,6 +264,215 @@ private static void ReplacementConnectionUsesSemaphoreTest(string connectionStri
+
+                Assert.True(internalConnection1.IsEnlistedInTransaction, "First connection not in transaction");
+                Assert.True(internalConnection1.IsTransactionRoot, "First connection not transaction root");
+                Assert.True(internalConnection2.IsEnlistedInTransaction, "Second connection not in transaction");
+                Assert.False(internalConnection2.IsTransactionRoot, "Second connection is transaction root");
+
+                // Attempt to re-use root connection
+                connection1.Close();
+                SqlConnection connection3 = new SqlConnection(connectionString);
+                connection3.Open();
+
+                Assert.True(connectionPool.ContainsConnection(connection3), "New connection in wrong pool");
+                Assert.True(internalConnection1.IsInternalConnectionOf(connection3), "Root connection was not re-used");
+
+                // Attempt to re-use non-root connection
+                connection2.Close();
+                SqlConnection connection4 = new SqlConnection(connectionString);
+                connection4.Open();
+                Assert.True(internalConnection2.IsInternalConnectionOf(connection4), "Connection did not re-use expected internal connection");
+                Assert.True(connectionPool.ContainsConnection(connection4), "New connection is in the wrong pool");
+                connection4.Close();
+
+                // Use a different connection string
+                SqlConnection connection5 = new SqlConnection(connectionString + ";App=SqlConnectionPoolUnitTest;");
+                connection5.Open();
+                Assert.False(internalConnection2.IsInternalConnectionOf(connection5), "Connection with different connection string re-used internal connection");
+                Assert.False(connectionPool.ContainsConnection(connection5), "Connection with different connection string is in same pool");
+                connection5.Close();
+
+                transScope.Complete();
+            }
+
+            Assert.Equal(2, connectionPool.ConnectionCount);
+        }
+
+        /// <summary>
+        /// Checks that connections in the transaction pool are not cleaned out, and the root transaction is put into "stasis" when it ages
+        /// </summary>
+        /// <param name="connectionString"></param>
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        [ClassData(typeof(ConnectionPoolConnectionStringProvider))]
+        public static void TransactionCleanupTest(string connectionString)
+        {
+            SqlConnection.ClearAllPools();
+            ConnectionPoolWrapper connectionPool = null;
+
+            using (TransactionScope transScope = new TransactionScope())
+            {
+                SqlConnection connection1 = new SqlConnection(connectionString);
+                SqlConnection connection2 = new SqlConnection(connectionString);
```

Comment:
Using will "dispose" resources properly.
```suggestion
                using SqlConnection connection1 = new SqlConnection(connectionString);
                using SqlConnection connection2 = new SqlConnection(connectionString);
```